_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,107 @@
+
+<br/>
+> Blessed are the young, for they shall inherit the national debt. - Herbet Hoover
+
+<br/>
+Maintaining a body of software becomes increasingly difficult as it ages and becomes larger in size. This is further aggravated by the reality that software engineers are not tethered to their creations. They leave and new engineers takes their place. New engineers need to understand the legacy left by the predecessors sans the education of having built the system in the first place. It is unreasonable to expect that a new engineer is going to be as effective as the seasoned veteran. We recognise nobody joins a new company with knowledge of its internal processes, politics and three-letter-acryonyms, and so we should expect the same of its software systems. Reducing the amount of learning required to be effective is a suitable means to ensuring that turnover does not effect the mean-time-to-repair and change-lead-time. Both the AWS and Google approaches to address this; two-pizza-teams place bounds around the amount of service-related knowledge any team member would be required to know, and SRE practices ensure teams do not become encumbered by poorly operated services.
+
+<br/>
+<center><img src="/img/tech-debt.png" width="50%"/></center>
+
+<br/>
+Outsourcing to a third party could be considered a close relative to dampening the effects of technical debt through smaller teams and specialisation. Cloud computing is the leading trend in outsourcing today. Hosting, databases, queues, and more are just a HTTP request away. The architectures that were once the territory of companies with a cast of thousands are now within the reach of a trio of computer science dropouts. Technical debt is still amassed when relying on services from a cloud provider: there is always the risk a service will be deprecated, or that it will be modified, or that it will no longer be suitable for our needs. But this is significantly smaller than the debt accrued when hosting the service ourselves. We would become responsible for the implementation, for maintenance, for the politics of prioritising new features among various internal stakeholders, and be held accountable for when the service inevitably fails. This a collosal amount of nonsense to suffer for a service that does not grant competitive advantage. If technical debt is inevitable (and I believe it is), I would rather acquire it in service of providing functions that are critical to business goals.
```

Comment:
Hmm... need to sharpen the terminology here. "Outsourcing to a third party ... close relative etc" .
Outsourcing has a very specific meaning. 
If I fire all my developers and hand the contract to TCS, that is outsourcing. Don't think that will help tech debt. 
How about "Relying on specialised  expert providers for specific managed services could be considered a close..."

Reply:
"collosal" should be seplt "colossal" - former sounds like some kind of rectal exam